klog_test.go

```diff
@@ -35,11 +35,24 @@ import (
 	"time"
 
 	"github.com/go-logr/logr"
+	"go.uber.org/goleak"
```

Comment:
We can't have `go.uber.org/goleak` as dependency of klog itself.

You can create an `examples/flushing` directory where you provide a `main.go` file which shows how to use the new API and a test that uses `goleak`.

Reply:
Sure, I already thought so. I'll move this in into a separate directory.